### ISSUE TITLE
Get all our php files showing up with syntax highlighting here on Github

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#4404]
**Description of the need**

Syntax highlighting is a nice feature when viewing code, but per default Github doesn't show some of Backdrop's typical php file extensions highlighted (.module, .install, .test).

**Proposed solution**

It's easy to tell the Github highlighter, which files are also supposed to be php with a **.gitattributes** file.
See an example in this [contrib project](https://github.com/backdrop-contrib/i18n/blob/1.x-1.x/.gitattributes)

**Alternatives that have been considered**

Having syntax highlighting for all our php files here on Github is a nice-to-have thing, but not really essential.

What do people think?

**Comment by quicksketch** [comment]
I didn't know that was possible. Neat! Last I checked GitHub didn't have an option to support arbitrary file extensions. Can you provide a link to the documentation for that GitHub feature?

We also now have the ability to exclude files from the release archives when the packager runs. We could exclude the `.gitattributes` file if it's something we don't want in the archives.

**Comment by indigoxela** [comment]
> Can you provide a link to the documentation for that GitHub feature?

Sure, here's the [README](https://github.com/github/linguist#using-gitattributes)

**Comment by indigoxela** [comment]
Sorry, almost forgot about this - here we go with a [PR](https://github.com/backdrop/backdrop/pull/3194)

**Comment by indigoxela** [comment]
"Testing" is super easy - compare the display of the same file...

With .gitattributes:

https://github.com/indigoxela/backdrop/blob/syntax-highlight-github/core/modules/admin_bar/admin_bar.module

Without it:

https://github.com/backdrop/backdrop/blob/1.16.x/core/modules/admin_bar/admin_bar.module

**Comment by klonos** [comment]
Thanks @indigoxela 👍 ...I've added a comment in the PR about including `.profile` files as php as well. Other than that, this is RTBC.

**Comment by laryn** [comment]
They show up the same in my browser -- is it a setting I need to activate?

![Screen Shot 2020-07-23 at 12 02 55 PM](https://user-images.githubusercontent.com/871421/88315829-7a015300-ccdc-11ea-93b5-63024bfd89a9.jpg)

![Screen Shot 2020-07-23 at 12 03 02 PM](https://user-images.githubusercontent.com/871421/88315840-7ec60700-ccdc-11ea-9272-5f3b69953efb.jpg)

**Comment by klonos** [comment]
Hmm 🤔 ...the coding highlighting showed up for me just few mins ago 🤷 

...did my adding a comment in the PR cause this bug somehow? 😅

**Comment by klonos** [comment]
...OK, something weird is happening ...I removed my comment with the code suggestion from the PR (I know, far-fetched, but just for the sake of testing) ...and then the highlighting did show. I then refreshed the page, and it was gone. Caching bug? 🤷

**Comment by klonos** [comment]
...yup, kept refreshing the page, and the highlighting comes and goes 😅 ...I'm sure that GitHub will eventually get to it.

**Comment by klonos** [comment]
...still does not change my code review and RTBC (with the minor addition of `.profile` files).

**Comment by indigoxela** [comment]
@laryn both, Github caching and webbrowser caching can be misleading here. For me it works to simply refresh (F5).

@klonos .profile have been added to the list:

https://github.com/indigoxela/backdrop/blob/syntax-highlight-github/core/profiles/standard/standard.profile

**Comment by laryn** [comment]
Looks good to me now. 👍

**Comment by quicksketch** [comment]
Thanks folks! I merged https://github.com/backdrop/backdrop/pull/3194 into 1.x and 1.16.x. I also excluded `.gitattributes` from being included in packages downloaded from BackdropCMS.org, where they might otherwise conflict with files added by developers on their own local projects.

https://github.com/backdrop-ops/backdropcms.org/commit/b4216fcc63ac38d4a044ec1454e5bbf046714889 (botched this one)
https://github.com/backdrop-ops/backdropcms.org/commit/af00c1b324a72d7cb3879210b7feb9fc87a9a46d (fix for the previous commit)
